src/cluster/distributed-state-registry.js

```diff
@@ -180,10 +180,15 @@ module.exports = class DistributedStateRegistry extends EventEmitter {
         nodes: {},
         checkSum: this._createCheckSum(name)
       }
-      this.emit('add', name)
     }
 
+    const exists = this._data[name].nodes[serverName]
+
     this._data[name].nodes[serverName] = true
+
+    if (exists) {
+      this.emit('add', name, serverName)
```

Comment:
Will do. Will also take this PR over the fix you made yesterday. Thanks for raising!